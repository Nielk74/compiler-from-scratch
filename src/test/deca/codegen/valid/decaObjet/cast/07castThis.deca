// Description:
//    Programme qui teste le cast de this
//
// Resultats:
//    07castThis.expected
//
// Historique:
//    cree le 16/06/2022
class A {
    int x = 10;
}

class B extends A {
    float x = ((A)(this)).x * 2.5;
}

{
    B b = new B();
    println("b.x = " ,b.x);
}