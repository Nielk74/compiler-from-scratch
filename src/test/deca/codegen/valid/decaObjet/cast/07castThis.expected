b.x = 2.50000e+01
